### PR TITLE
Onboard Package Validation SDK

### LINKED ISSUE #33981: Use Package Validation SDK
https://devblogs.microsoft.com/dotnet/package-validation/

Suppressions: https://github.com/dotnet/sdk/blob/18b394cf392030683ec0e7abd2ce588803d8d514/src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/SuppressionTests/SuppressionEngineTests.cs#L139

Baseline Suppressions: https://github.com/dotnet/sdk/blob/18b394cf392030683ec0e7abd2ce588803d8d514/src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/SuppressionTests/SuppressionEngineTests.cs#L190

`DisablePackageBaselineValidation` to disable baseline validation in a given project (we want this to be conditioned on `IsServicingBuild`)

https://github.com/dotnet/msbuild/issues/4249 for conditionally using SDK tags in a project

### PR DESCRIPTION
**Pull request by wtgodbe**
Fixes https://github.com/dotnet/aspnetcore/issues/33981

Still a couple outstanding questions - 

- [ ] How/When can we get this version updated automatically? I believe it comes from dotnet/sdk right now - we could add a back-edge subscription that only fires once a week or so?
- [ ] What else should we do to validate that this is doing what we expect? I see the below in a binlog of one of our shipping projects:

PackVal

### CODE REVIEW

**Review by dougbu — CHANGES_REQUESTED** [review]
We can't use this SDK until it no longer worsens the file contention issues in our build. Suspect it works in dotnet/runtime only because they use the Microsoft.Build.Traversal SDK.

File: Directory.Build.targets [hunk 1]
```diff
@@ -66,6 +66,12 @@
          '$(IsBenchmarkProject)' == 'true' OR '$(IsSampleProject)' == 'true' OR '$(IsMicrobenchmarksProject)' == 'true') ">false</IsPackable>
   </PropertyGroup>
 
+  <!-- Properties for Package Validation -->
+  <!-- We only want to check for API changes in servicing branches -->
+  <PropertyGroup Condition=" '$(IsServicingBuild)' != 'true' ">
+    <DisablePackageBaselineValidation>true</DisablePackageBaselineValidation>
```
Comment (dougbu) [inline Directory.Build.targets:72]: Don't we need to set properties somewhere to inform Microsoft.DotNet.PackageValidation of the baseline package version once we _are_ in servicing❔

**Comment by Anipik** [comment]
> How/When can we get this version updated automatically? I believe it comes from dotnet/sdk right now - we could add a back-edge subscription that only fires once a week or so?

i am planning to make it a part of sdk before preview 7 change. so maybe waiting here is the best option

File: Directory.Build.targets (hunk 1 above)
Comment (Anipik) [reply Directory.Build.targets:72]: correct we dont need this, baseline validation wont run by default

**Comment by wtgodbe** [comment]
@Anipik we're concerned about this going in to the dotnet SDK if it winds up causing these build races for us - it might block us from updating to a newer SDK.

@rainersigwald we're seeing more instances of the file contention issue here, but on different projects:

> D:\workspace\_work\1\s\eng\AfterSolutionBuild.targets(15,5): error MSB4018: System.IO.IOException: The process cannot access the file 'D:\workspace\_work\1\s\artifacts\packages\Release\Shipping\Microsoft.AspNetCore.App.Ref.6.0.0-ci.nupkg' because it is being used by another process. [D:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\6.0.0-beta.21329.1\tools\AfterSolutionBuild.proj]

How can we help investigate?

**Comment by dougbu** [comment]
I doubt it'll help but let's see if @TanayParikh's targeted fix (#34137) does the trick. Might need something similar for other file contention issues this PR brings to the forefront.

**Comment by dougbu** [comment]
> How can we help investigate?

See Build channel in Teams for discussion of prior build contention investigations. Starting point is a binary log for whatever job looks worst to you.

**Comment by TanayParikh** [comment]
```
eng/AfterSolutionBuild.targets(15,5): error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "RepoTasks.RemoveSharedFrameworkDependencies" task failed unexpectedly.
System.IO.IOException: The process cannot access the file '/datadisks/disk1/workspace/_work/1/s/artifacts/packages/Release/Shipping/Microsoft.AspNetCore.App.Runtime.linux-x64.6.0.0-ci.nupkg' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Init(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) in System.Private.CoreLib.dll:token 0x60000ce+0x0
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) in System.Private.CoreLib.dll:token 0x60000cc+0x31
   at System.IO.Strategies.Net5CompatFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize) in System.Private.CoreLib.dll:token 0x600601b+0x6f
   at System.IO.File.Open(String path, FileMode mode) in System.Private.CoreLib.dll:token 0x6005c1d+0x0
   at RepoTasks.RemoveSharedFrameworkDependencies.FilterDependencies(String targetPath, ISet`1 dependencyToRemove) in /_/eng/tools/RepoTasks/RemoveSharedFrameworkDependencies.cs:line 48
   at RepoTasks.RemoveSharedFrameworkDependencies.Execute() in /_/eng/tools/RepoTasks/RemoveSharedFrameworkDependencies.cs:line 38
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() in Microsoft.Build.dll:token 0x60015fc+0x3e
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) in Microsoft.Build.dll:token 0x600147b+0x804
```

https://github.com/dotnet/aspnetcore/issues/32219

**Pull request closed without merging**